### ISSUE TITLE
New-AzBatchPool fails if $context is from Get-AzBatchAccountKeys due to missing assembly Microsoft.AspNetCore.WebUtilities

### ISSUE THREAD

**Issue by MHHenriksen** [Azure/azure-powershell#7921]
### Description

New-AzBatchPool fails if $context is from Get-AzBatchAccountKeys due to missing assembly Microsoft.AspNetCore.WebUtilities.

Workarounds:
- Use Get-AzBatchAccount instead
- Download the Microsoft.AspNetCore.WebUtilities package from Nuget and add the assembly to the Az.Batch module directory

Exception Details:
```
Message        : Could not load file or assembly 'Microsoft.AspNetCore.WebUtilities, Version=1.0.2.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'. The system cannot find the file s
                 pecified.
StackTrace     :    at Microsoft.Azure.Batch.Protocol.BatchSharedKeyCredential.ProcessHttpRequestAsync(HttpRequestMessage httpRequest, CancellationToken cancellationToken)
                    at Microsoft.Azure.Batch.Protocol.PoolOperations.AddWithHttpMessagesAsync(PoolAddParameter pool, PoolAddOptions poolAddOptions, Dictionary`2 customHeaders, Cancellati
                 onToken cancellationToken)
                    at Microsoft.Azure.Batch.Protocol.BatchRequestBase`2.ExecuteRequestWithCancellationAsync(Func`2 func)
                    at Microsoft.Azure.Batch.Protocol.BatchRequestBase`2.ExecuteRequestAsync()
                    at Microsoft.Azure.Batch.ProtocolLayer.ProcessAndExecuteBatchRequest[TResponse](IBatchRequest`1 request, BehaviorManager bhMgr)
                    at Microsoft.Azure.Batch.CloudPool.CommitAsync(IEnumerable`1 additionalBehaviors, CancellationToken cancellationToken)
                    at Microsoft.Azure.Batch.UtilitiesInternal.WaitAndUnaggregateException(Task task, IEnumerable`1 rootBehaviors, IEnumerable`1 additionalBehaviors)
                    at Microsoft.Azure.Batch.CloudPool.Commit(IEnumerable`1 additionalBehaviors)
                    at Microsoft.Azure.Commands.Batch.Models.BatchClient.CreatePool(NewPoolParameters parameters)
                    at Microsoft.Azure.Commands.Batch.NewBatchPoolCommand.ExecuteCmdlet()
                    at Microsoft.WindowsAzure.Commands.Utilities.Common.AzurePSCmdlet.ProcessRecord()
Exception      : System.IO.FileNotFoundException
InvocationInfo : {New-AzBatchPool}
Line           :     New-AzBatchPool -Id $PoolName -VirtualMachineSize "standard_a2" -VirtualMachineConfiguration $configuration -AutoScaleFormula $AutoScaleFormula -ApplicationPackageRe
                 ferences $ApplicationPackageReferences -AutoScaleEvaluationInterval $Timespan -MaxTasksPerComputeNode 5 -StartTask $StartTask -BatchContext $Context | Out-Null;

Position       : At C:\code\tools\terraform\scripts\azr_batch.ps1:126 char:5
                 +     New-AzBatchPool -Id $PoolName -VirtualMachineSize "standard_a2" - ...
                 +     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

### Script/Steps for Reproduction

Any of the examples at https://docs.microsoft.com/en-us/powershell/module/azurerm.batch/new-azurebatchpool?view=azurermps-6.13.0 BUT using $Context = Get-AzBatchAccountKeys 

```powershell
$Context = Get-AzBatchAccountKeys
$configuration = New-Object -TypeName "Microsoft.Azure.Commands.Batch.Models.PSCloudServiceConfiguration" -ArgumentList @(4,"*")
New-AzBatchPool -Id "MyPool" -VirtualMachineSize "Small" -CloudServiceConfiguration $configuration  -TargetDedicatedComputeNodes 3 -BatchContext $Context
```

### Module Version

```powershell
Get-Module -ListAvailable

    Directory: C:\Users\mhh\Documents\PowerShell\Modules


ModuleType Version    Name                                PSEdition ExportedCommands
---------- -------    ----                                --------- ----------------
Script     0.6.1      Az.Aks                              Core,Desk {Get-AzAks, New-AzAks, Remove-AzAks, Import-AzAksCredential...}
Script     0.6.1      Az.AnalysisServices                 Core,Desk {Resume-AzAnalysisServicesServer, Suspend-AzAnalysisServicesServer, Get-AzAnalysisServicesServer, Remove-AzAnalysis...
Script     0.6.1      Az.ApiManagement                    Core,Desk {Add-AzApiManagementRegion, Get-AzApiManagementSsoToken, New-AzApiManagementHostnameConfiguration, New-AzApiManagem...
Script     0.6.1      Az.ApplicationInsights              Core,Desk {Get-AzApplicationInsights, New-AzApplicationInsights, Remove-AzApplicationInsights, Set-AzApplicationInsightsPrici...
Script     0.6.1      Az.Automation                       Core,Desk {Get-AzAutomationHybridWorkerGroup, Remove-AzAutomationHybridWorkerGroup, Get-AzAutomationJobOutputRecord, Import-A...
Script     0.6.1      Az.Batch                            Core,Desk {Remove-AzBatchAccount, Get-AzBatchAccount, Get-AzBatchAccountKeys, New-AzBatchAccount...}
Script     0.6.1      Az.Billing                          Core,Desk {Get-AzBillingInvoice, Get-AzBillingPeriod, Get-AzEnrollmentAccount}
Script     0.6.1      Az.Cdn                              Core,Desk {Get-AzCdnProfile, Get-AzCdnProfileSsoUrl, New-AzCdnProfile, Remove-AzCdnProfile...}
Script     0.6.1      Az.CognitiveServices                Core,Desk {Get-AzCognitiveServicesAccount, Get-AzCognitiveServicesAccountKey, Get-AzCognitiveServicesAccountSkus, Get-AzCogni...
Script     0.6.1      Az.Compute                          Core,Desk {Remove-AzAvailabilitySet, Get-AzAvailabilitySet, New-AzAvailabilitySet, Update-AzAvailabilitySet...}
Script     0.6.1      Az.Consumption                      Core,Desk {Get-AzConsumptionBudget, Get-AzConsumptionMarketplace, Get-AzConsumptionPriceSheet, Get-AzConsumptionReservationDe...
Script     0.6.1      Az.ContainerInstance                Core,Desk {New-AzContainerGroup, Get-AzContainerGroup, Remove-AzContainerGroup, Get-AzContainerInstanceLog}
Script     0.6.1      Az.ContainerRegistry                Core,Desk {New-AzContainerRegistry, Get-AzContainerRegistry, Update-AzContainerRegistry, Remove-AzContainerRegistry...}
Script     0.6.1      Az.DataLakeAnalytics                Core,Desk {Get-AzDataLakeAnalyticsDataSource, New-AzDataLakeAnalyticsCatalogCredential, Remove-AzDataLakeAnalyticsCatalogCred...
Script     0.6.1      Az.DataLakeStore                    Core,Desk {Get-AzDataLakeStoreTrustedIdProvider, Remove-AzDataLakeStoreTrustedIdProvider, Remove-AzDataLakeStoreFirewallRule,...
Script     0.6.1      Az.DevTestLabs                      Core,Desk {Get-AzDtlAllowedVMSizesPolicy, Get-AzDtlAutoShutdownPolicy, Get-AzDtlAutoStartPolicy, Get-AzDtlVMsPerLabPolicy...}
Script     0.6.1      Az.Dns                              Core,Desk {Get-AzDnsRecordSet, New-AzDnsRecordConfig, Remove-AzDnsRecordSet, Set-AzDnsRecordSet...}
Script     0.6.1      Az.EventGrid                        Core,Desk {New-AzEventGridTopic, Get-AzEventGridTopic, Set-AzEventGridTopic, New-AzEventGridTopicKey...}
Script     0.6.1      Az.EventHub                         Core,Desk {New-AzEventHubNamespace, Get-AzEventHubNamespace, Set-AzEventHubNamespace, Remove-AzEventHubNamespace...}
Script     0.6.1      Az.Insights                         Core,Desk {Get-AzMetricDefinition, Get-AzMetric, Remove-AzLogProfile, Get-AzLogProfile...}
Script     0.6.1      Az.IotHub                           Core,Desk {Add-AzIotHubKey, Get-AzIotHubEventHubConsumerGroup, Get-AzIotHubConnectionString, Get-AzIotHubJob...}
Script     0.6.1      Az.KeyVault                         Core,Desk {Add-AzKeyVaultCertificate, Update-AzKeyVaultCertificate, Stop-AzKeyVaultCertificateOperation, Get-AzKeyVaultCertif...
Script     0.6.1      Az.LogicApp                         Core,Desk {Get-AzIntegrationAccountAgreement, Get-AzIntegrationAccountCallbackUrl, Get-AzIntegrationAccountCertificate, Get-A...
Script     0.6.1      Az.MachineLearning                  Core,Desk {Move-AzMlCommitmentAssociation, Get-AzMlCommitmentAssociation, Get-AzMlCommitmentPlanUsageHistory, Remove-AzMlComm...
Script     0.6.1      Az.MachineLearningCompute           Core,Desk {Get-AzMlOpCluster, Get-AzMlOpClusterKey, Test-AzMlOpClusterSystemServicesUpdateAvailability, Update-AzMlOpClusterS...
Script     0.6.1      Az.MarketplaceOrdering              Core,Desk {Get-AzMarketplaceTerms, Set-AzMarketplaceTerms}
Script     0.6.1      Az.Media                            Core,Desk {Sync-AzMediaServiceStorageKeys, Set-AzMediaServiceKey, Get-AzMediaServiceKeys, Get-AzMediaServiceNameAvailability...}
Script     0.6.1      Az.Network                          Core,Desk {Add-AzApplicationGatewayAuthenticationCertificate, Get-AzApplicationGatewayAuthenticationCertificate, New-AzApplic...
Script     0.6.1      Az.NotificationHubs                 Core,Desk {Get-AzNotificationHub, Get-AzNotificationHubAuthorizationRules, Get-AzNotificationHubListKeys, Get-AzNotificationH...
Script     0.6.1      Az.OperationalInsights              Core,Desk {New-AzOperationalInsightsAzureActivityLogDataSource, New-AzOperationalInsightsCustomLogDataSource, Disable-AzOpera...
Script     0.6.1      Az.PolicyInsights                   Core,Desk {Get-AzPolicyEvent, Get-AzPolicyState, Get-AzPolicyStateSummary, Get-AzPolicyRemediation...}
Script     0.6.1      Az.PowerBIEmbedded                  Core,Desk {Remove-AzPowerBIWorkspaceCollection, Get-AzPowerBIWorkspaceCollection, Get-AzPowerBIWorkspaceCollectionAccessKeys,...
Script     0.6.1      Az.Profile                          Core,Desk {Disable-AzDataCollection, Disable-AzContextAutosave, Enable-AzDataCollection, Enable-AzContextAutosave...}
Script     0.6.1      Az.RecoveryServices                 Core,Desk {Get-AzRecoveryServicesBackupProperty, Get-AzRecoveryServicesVault, Get-AzRecoveryServicesVaultSettingsFile, New-Az...
Script     0.6.1      Az.RedisCache                       Core,Desk {Remove-AzRedisCachePatchSchedule, New-AzRedisCacheScheduleEntry, Get-AzRedisCachePatchSchedule, New-AzRedisCachePa...
Script     0.6.1      Az.Relay                            Core,Desk {New-AzRelayNamespace, Get-AzRelayNamespace, Set-AzRelayNamespace, Remove-AzRelayNamespace...}
Script     0.6.1      Az.Resources                        Core,Desk {Get-AzProviderOperation, Remove-AzRoleAssignment, Get-AzRoleAssignment, New-AzRoleAssignment...}
Script     0.6.1      Az.ServiceBus                       Core,Desk {New-AzServiceBusNamespace, Get-AzServiceBusNamespace, Set-AzServiceBusNamespace, Remove-AzServiceBusNamespace...}
Script     0.6.1      Az.ServiceFabric                    Core,Desk {Add-AzServiceFabricApplicationCertificate, Add-AzServiceFabricClientCertificate, Add-AzServiceFabricClusterCertifi...
Script     0.6.1      Az.SignalR                          Core,Desk {New-AzSignalR, Get-AzSignalR, Get-AzSignalRKey, New-AzSignalRKey...}
Script     0.6.1      Az.Sql                              Core,Desk {Get-AzSqlDatabaseTransparentDataEncryption, Get-AzSqlDatabaseTransparentDataEncryptionActivity, Set-AzSqlDatabaseT...
Script     0.6.1      Az.Storage                          Core,Desk {Get-AzStorageAccount, Get-AzStorageAccountKey, New-AzStorageAccount, New-AzStorageAccountKey...}
Script     0.6.1      Az.StreamAnalytics                  Core,Desk {Get-AzStreamAnalyticsFunction, Get-AzStreamAnalyticsDefaultFunctionDefinition, New-AzStreamAnalyticsFunction, Remo...
Script     0.6.1      Az.Tags                             Core,Desk {Remove-AzTag, Get-AzTag, New-AzTag}
Script     0.6.1      Az.TrafficManager                   Core,Desk {Add-AzTrafficManagerCustomHeaderToEndpoint, Remove-AzTrafficManagerCustomHeaderFromEndpoint, Add-AzTrafficManagerC...
Script     0.6.1      Az.UsageAggregates                  Core,Desk Get-UsageAggregates
Script     0.6.1      Az.Websites                         Core,Desk {Get-AzAppServicePlan, Set-AzAppServicePlan, New-AzAppServicePlan, Remove-AzAppServicePlan...}
Script     1.3.1      Configuration                       Desk      {Import-Configuration, Export-Configuration, Get-StoragePath, Add-MetadataConverter...}
Script     2.0.230    oh-my-posh                          Desk      {Show-Colors, Show-ThemeColors, Show-ThemeSymbols, Get-Theme...}
Script     1.2.1      Pansies                             Desk      {New-Text, Write-Host, Get-Gradient, Get-Complement...}
Script     0.7.3      posh-git                            Desk      {Invoke-NullCoalescing, Add-PoshGitToProfile, Get-PromptPath, Write-GitStatus...}
Script     3.0.3      PowerLine                           Desk      {Set-PowerLinePrompt, Export-PowerLinePrompt, Add-PowerLineBlock, Remove-PowerLineBlock...}
Script     1.0.0      WindowsPSModulePath                 Core,Desk Add-WindowsPSModulePath


    Directory: C:\program files\powershell\6\Modules


ModuleType Version    Name                                PSEdition ExportedCommands
---------- -------    ----                                --------- ----------------
Manifest   6.1.0.0    CimCmdlets                          Core      {Get-CimAssociatedInstance, Get-CimClass, Get-CimInstance, Get-CimSession...}
Manifest   1.1.0.0    Microsoft.PowerShell.Archive        Desk      {Compress-Archive, Expand-Archive}
Manifest   6.1.0.0    Microsoft.PowerShell.Diagnostics    Core      {Get-WinEvent, New-WinEvent}
Manifest   6.1.0.0    Microsoft.PowerShell.Host           Core      {Start-Transcript, Stop-Transcript}
Manifest   6.1.0.0    Microsoft.PowerShell.Management     Core      {Add-Content, Clear-Content, Clear-ItemProperty, Join-Path...}
Manifest   6.1.0.0    Microsoft.PowerShell.Security       Core      {Get-Acl, Set-Acl, Get-PfxCertificate, Get-Credential...}
Manifest   6.1.0.0    Microsoft.PowerShell.Utility        Core      {Format-List, Format-Custom, Format-Table, Format-Wide...}
Manifest   6.1.0.0    Microsoft.WSMan.Management          Core      {Disable-WSManCredSSP, Enable-WSManCredSSP, Get-WSManCredSSP, Set-WSManQuickConfig...}
Script     1.1.7.2    PackageManagement                   Desk      {Find-Package, Get-Package, Get-PackageProvider, Get-PackageSource...}
Script     1.6.7      PowerShellGet                       Desk      {Find-Command, Find-DSCResource, Find-Module, Find-RoleCapability...}
Script     0.0        PSDesiredStateConfiguration         Desk      {Get-PSCurrentConfigurationNode, Get-DscResource, Write-Log, ValidateNodeExclusiveResources...}
Script     6.1.0.0    PSDiagnostics                       Core      {Disable-PSTrace, Disable-PSWSManCombinedTrace, Disable-WSManTrace, Enable-PSTrace...}
Script     2.0.0      PSReadLine                          Desk      {Get-PSReadLineKeyHandler, Set-PSReadLineKeyHandler, Remove-PSReadLineKeyHandler, Get-PSReadLineOption...}
Binary     1.1.2      ThreadJob                           Desk      Start-ThreadJob

```

### Environment Data

```powershell
$PSVersionTable

Name                           Value
----                           -----
PSVersion                      6.1.0
PSEdition                      Core
GitCommitId                    6.1.0
OS                             Microsoft Windows 10.0.17134
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Debug Output

```
DEBUG: 14:15:41 - NewBatchPoolCommand begin processing with ParameterSet 'VirtualMachineAndTargetDedicated'.
DEBUG: 14:15:41 - using account id '*snip*'...
DEBUG: [Common.Authentication]: Authenticating using Account: '*snip*', environment: 'AzureCloud', tenant: '*snip*'
DEBUG: [Common.Authentication]: Authenticating using Account: '*snip*', environment: 'AzureCloud', tenant: '*snip*'
New-AzBatchPool : Could not load file or assembly 'Microsoft.AspNetCore.WebUtilities, Version=1.0.2.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'. The system cannot find the file specified.
At line:1 char:1
+ New-AzBatchPool -id mypool3 -VirtualMachineSize Standard_A1_v2 -Virtu ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : CloseError: (:) [New-AzBatchPool], FileNotFoundException
+ FullyQualifiedErrorId : Microsoft.Azure.Commands.Batch.NewBatchPoolCommand

DEBUG: AzureQoSEvent: CommandName - New-AzBatchPool; IsSuccess - False; Duration - 00:00:00.0284884; Exception - System.IO.FileNotFoundException: Could not load file or assembly 'Microsoft.AspNetCore.WebUtilities, Version=1.0.2.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'. The system cannot find the file specified.
File name: 'Microsoft.AspNetCore.WebUtilities, Version=1.0.2.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'
   at Microsoft.Azure.Batch.Protocol.BatchSharedKeyCredential.ProcessHttpRequestAsync(HttpRequestMessage httpRequest, CancellationToken cancellationToken)
   at Microsoft.Azure.Batch.Protocol.PoolOperations.AddWithHttpMessagesAsync(PoolAddParameter pool, PoolAddOptions poolAddOptions, Dictionary`2 customHeaders, CancellationToken cancellationToken)
   at Microsoft.Azure.Batch.Protocol.BatchRequestBase`2.ExecuteRequestWithCancellationAsync(Func`2 func)
   at Microsoft.Azure.Batch.Protocol.BatchRequestBase`2.ExecuteRequestAsync()
   at Microsoft.Azure.Batch.ProtocolLayer.ProcessAndExecuteBatchRequest[TResponse](IBatchRequest`1 request, BehaviorManager bhMgr)
   at Microsoft.Azure.Batch.CloudPool.CommitAsync(IEnumerable`1 additionalBehaviors, CancellationToken cancellationToken)
   at Microsoft.Azure.Batch.UtilitiesInternal.WaitAndUnaggregateException(Task task, IEnumerable`1 rootBehaviors, IEnumerable`1 additionalBehaviors)
   at Microsoft.Azure.Batch.CloudPool.Commit(IEnumerable`1 additionalBehaviors)
   at Microsoft.Azure.Commands.Batch.Models.BatchClient.CreatePool(NewPoolParameters parameters)
   at Microsoft.Azure.Commands.Batch.NewBatchPoolCommand.ExecuteCmdlet()
   at Microsoft.WindowsAzure.Commands.Utilities.Common.AzurePSCmdlet.ProcessRecord()

;
DEBUG: Finish sending metric.
DEBUG: 14:15:41 - NewBatchPoolCommand end processing.
DEBUG: 14:15:41 - NewBatchPoolCommand end processing.
```

**Comment by cormacpayne** [comment]
@matthchr Hey Matt, would you mind taking a look at this issue?

**Comment by matthchr** [comment]
This is fixed in `1.0.1-preview` of the `Az.Batch` module.

You can acquire it by running the following:
`Install-Module -Name Az.Batch -Repository PSGallery -AllowPrerelease -Force`
